packages/node/src/transports/http.ts

```diff
@@ -85,6 +103,17 @@ function createRequestExecutor(
   const { hostname, pathname, port, protocol, search } = new URL(options.url);
   return function makeRequest(request: TransportRequest): Promise<TransportMakeRequestResponse> {
     return new Promise((resolve, reject) => {
+      let bodyStream = streamFromBody(request.body);
+
+      if (request.body.length > GZIP_THRESHOLD) {
+        options.headers = {
+          ...options.headers,
+          'content-encoding': 'gzip',
+        };
+
+        bodyStream = bodyStream.pipe(createGzip());
```

Comment:
I think it makes sense to extract these transport changes into a separate PR, what do you think?

Reply:
Actually yes. The types made it look like `end` only look a string (`req.end(chunk: string)`) but I've just noticed these are custom types from `http-module.ts`!